### PR TITLE
Use search params on filtering values

### PR DESCRIPTION
**Pull request by marttsang2**
In searching company name and job title, i use key press the enter button to submit instead of debounce input.

In the search pop-up component, search params only apply when the user clicks the confirm button.

### CODE REVIEW

**Comment by Pjaijai** [comment]
Currently the design do not show the `確定` button. If we remove the debounce thing, user need to click the filter to click `確定` or press 'Enter'. Can we keep debounce ? Or we making the `確定` button out. 
 
Great Job ! Definitely solving one of our main problem

**Comment by marttsang2** [comment]
Currently, I only make a confirm button for the search pop-up component. I can make the confirm button out if you want.

**Comment by Pjaijai** [comment]
@marttsang2  Please go ahead. As most users may not be tech-savvy, it's better to place the button out. Many thanks !!

**Comment by Pjaijai** [comment]
I will to the solve the file conflict after your work. Let me handle it.

**Comment by Pjaijai** [comment]
@marttsang2  One more thing. Please add your self into contributors page thanks

**Comment by marttsang2** [comment]
@Pjaijai updated the contributor list and moved the confirm button out of the search pop-up component. You can take a look. Thanks.

**Review by Pjaijai — APPROVED** [review]
Awesome work !

File: client/hooks/api/referral/search-referral.tsx [hunk 1]
```diff
@@ -89,25 +110,37 @@ const useSearchReferral = (type: ReferralType) => {
...
       // If it's negative or not a valid integer, set it to '0'
       setYoeMax("0")
+      createQueryString("yoeMax", "0")
+    }
+  }
+
+  const handleSubmitChange = () => {
+    router.push(pathname + "?" + params.toString())
+  }
+
```
[the remark this answers is not in the document]
Comment (marttsang2) [reply client/hooks/api/referral/search-referral.tsx:124]: I am using the search params string based on nextjs website. Of course, there are different ways to do the same thing. We can change it later if u think it is better.
https://nextjs.org/docs/app/api-reference/functions/use-search-params

**Pull request merged**